mesonbuild/build.py

```diff
@@ -3072,12 +3072,16 @@ class TestSetup:
     env: EnvironmentVariables
     exclude_suites: T.List[str]
 
-def get_sources_string_names(sources, backend):
+
+def get_sources_string_names(sources: T.Sequence[T.Union[str, BuildTarget, CustomTarget, CustomTargetIndex, GeneratedList, File, ExtractedObjects]],
```

Comment:
As usual, sequence means it would be legal to do:

```python
get_sources_string_names('foobar', current_backend)
```

But we already know that the only place this function is ever used, must pass a list and nothing but a list.